packages/wrangler/src/deploy/deploy.ts

```diff
@@ -742,6 +743,8 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 		if (props.dryRun) {
 			printBindings({ ...withoutStaticAssets, vars: maskedVars });
 		} else {
+			assert(accountId, "Missing accountId");
```

Comment:
This shouldn't be an assertion. If a user can get themselves in this situation, throw a UserError. An AssertionError is a mistake in our logic and will be sent to Sentry whereas a UserError won't.

Reply:
The assertion was just moved from the bottom to fix the type - it probably is something we want in Sentry tbf. Users _shouldn't_ be able to get to the deploy path without an account ID -- we should be prompting or accepting the config value

Reply:
I see, sorry I thought you introduced it. It should've been an UserError already. The config is already considered. It can only be undefined if not specified in args OR config.